service/src/main/java/edu/tamu/catalog/exception/NotFoundException.java

```diff
@@ -0,0 +1,7 @@
+package edu.tamu.catalog.exception;
+
+public class NotFoundException extends RuntimeException {
```

Comment:
This is not necessary as it never actually thrown.

Reply:
See above about query parameter `CatalogName`.

Reply:
This is not an appropriate exception for unsupported operation or not implemented.

Reply:
It is, however, consistent with the current/preexisting behavior.
That said, I am in favor of changing this behavior.